File: backend-java/src/test/java/com/skewwhiffy/auraltester/cucumber/InfoSteps.java
package com.skewwhiffy.auraltester.cucumber;

import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import lombok.AllArgsConstructor;
import lombok.val;
import org.springframework.http.HttpStatus;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

@AllArgsConstructor
public class InfoSteps {
    private final RestRequests restRequests;

    @Given("I hit the info endpoint")
    public void hitInfoEndpoint() {
        restRequests.makeGetRequest("/api/info");
    }

    @Then("system information is returned")
    public void systemInformationReturned() {
        val response = restRequests.getLastResponse();
        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
        val responseMap = restRequests.getLastResponseAsMap();
        assertThat(responseMap.get("version")).isInstanceOf(String.class);
    }
}

File: backend/src/test/kotlin/com/skewwhiffy/auraltester/cucumber/InfoSteps.kt
package com.skewwhiffy.auraltester.cucumber

import io.cucumber.java.en.Given
import io.cucumber.java.en.Then
import org.assertj.core.api.Assertions.assertThat
import org.springframework.http.HttpStatus

class InfoSteps(private val restRequests: RestRequests) {
    @Given("I hit the info endpoint")
    fun hitInfoEndpoint() {
        restRequests.makeGetRequest("/api/info")
    }

    @Then("system information is returned")
    fun systemInformationReturned() {
        val response = restRequests.lastResponse
        assertThat(response.status).isEqualTo(HttpStatus.OK.value())
        val responseMap = restRequests.lastResponseAsMap
        assertThat(responseMap["version"]).isInstanceOf(String::class.java)
    }
}